loadbalancer/src/main/java/com/boundary/dropwizard/consul/loadbalancer/AbstractLBFactory.java

```diff
@@ -72,20 +72,22 @@ protected ServiceHealthCache buildCache(Environment env, HealthClient healthClie
                 getWatchSeconds()
         );
 
-        env.lifecycle().manage(new Managed() {
-            @Override
-            public void start() throws Exception {
-                cache.start();
-                if (!cache.awaitInitialized(10, TimeUnit.SECONDS)) {
-                    throw new Exception("load balancer init timeout");
+        if (env != null) {
```

Comment:
this cache has to be started or it won't work - either ppl have to pass it in or we have to start it for them in the absence of an environment. once thing to do would be to make buildCache public so that it can be called separately and managed however, then passed provided in the createClient call, and move the environment management to somewhere more specific.